docs/architecture/domain.md

```diff
@@ -0,0 +1,125 @@
+## Domain Model
+
+![Definition](assets/definition-model.png)
+![Runtime](assets/runtime-model.png)
+
+### Source Code
+
+Source Code is a specified GitHub repository and branch that contains a `screwdriver.yaml` and the code required to build, test, and publish your application.
+
+### Step
+
+A step is a named action that needs to be performed, usually a single shell command. If the command finishes with a non-zero exit code, the step is considered a failure.
+
+### Container
+
+A container runs [steps] in an isolated environment. This is done in order to test the compatibility of code running in different environments with different versions, without affecting other [builds] that may be running at the same time. This is implemented using Docker containers.
+
+### Job
+
+A job consists of executing multiple sequential [steps] inside a specified [container]. If any step in the series fails, then the entire job is considered failed and subsequent steps will be skipped (unless configured otherwise).
+
+Jobs work by checking out the [source code] to a specified commit, setting the desired environment variables, and executing the specified [steps].
+
+During the job, the executing [steps] share three pieces of context:
+
+ - Filesystem
+ - [Container]
+ - [Metadata]
+
+Jobs are started by changes in [source code], as a trigger from the [workflow], or by human decision.
+
+#### Parallelization
+
+It is possible to parallelize a job by defining a matrix of environment variables. These are usually used for testing against multiple [containers] or test types.
+
+In this example job definition, 4 [builds] will run in parallel:
+```yaml
+image: node:((NODE_VERSION))
+steps:
+    test: npm run test-${TEST_TYPE}
+matrix:
+    NODE_VERSION:
+        - 4
+        - 6
+    TEST_TYPE:
+        - unit
+        - functional
+```
+
+ - `NODE_VERSION=4` and `TEST_TYPE=unit`
+ - `NODE_VERSION=4` and `TEST_TYPE=functional`
+ - `NODE_VERSION=6` and `TEST_TYPE=unit`
+ - `NODE_VERSION=6` and `TEST_TYPE=functional`
+
+### Build
+
+A build is an instance of a running [job]. All builds are assigned a unique build number. With a basic job configuration, only one build of a job will be running at any given time. If a [job matrix] is configured, then there can be multiple builds running in parallel.
+
+A build can be in one of five different states:
+
+ - `queued` - Build is waiting for available resources
+ - `running` - Build is actively running on an executor
```

Comment:
Should these match with the status when we do a get? https://github.com/screwdriver-cd/data-schema/blob/master/model/build.js#L51-L55

running -> in progress
successful -> success 
...